packages/suite-build/utils/alias.ts

```diff
@@ -1,27 +1,33 @@
 import * as path from 'path';
+import { A, D, pipe, S } from '@mobily/ts-belt';
+// we should not import files outside rootDir in TS composite project
+// @ts-ignore
+import tsConfig from '../../../tsconfig.aliases.json';
 
-// we cannot import file outside root dir as module in TS composite project
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const { compilerOptions } = require('../../../tsconfig.aliases.json');
+const {
+    compilerOptions: { paths },
+} = tsConfig;
 
-const { paths } = compilerOptions;
-type PathKey = keyof typeof paths;
+// Webpack resolve all aliases with wildcard automatically so we should remove /* and /index from end of paths
+const replacePathWildcard = S.replaceByRe(/(\/\*|\/index)$/, '');
+export const resolvePath = (relativePath: string) =>
+    path.resolve(__dirname, '..', '..', '..', relativePath);
 
-const getPath = (key: PathKey) => {
-    let p = paths[key][0];
-    if (p.endsWith('index')) {
-        p = p.slice(0, -5);
-    }
+export const prepareAliases = (aliasesPaths: Record<string, string[]>) =>
+    pipe(
+        aliasesPaths,
+        D.map(A.last),
+        D.toPairs,
```

Comment:
I had to look this one up, for example, turnt out it's simply `Object.entries()`.

Reply:
Just little bit different naming from FP world.` toPairs` is very common name used across all available FP libs.   Doing this with `Object.entries` would require doing imperative `for-of` loop, that force us to take care how exactly is object constructed which is another imperative stuff you don't want to take care about. It's not so significant for simpler things but more problem complex is, then more better declarative approach is.

In case there will be interest I can prepare something like intro/workshop about FP programming (with some focus on this lib we use). Show you imperative vs declarative, how and why are things better in FP world... It could be really beneficial for you because React as trying to teach us FP without even knowing it, but when you understand underlaying concepts it will makes you better also in React.

Reply:
Not sure why `for-of` would be needed. 

```
Object.entries({ key: 'value' }).map(([key, value]) => console.log(key, value));
```

But yes, I think we should all agree on one approach, just like with Redux, which we discussed a while back :) I am not against FP, it's just that we have clashing approaches and people who haven't seen you PRs don't even know about it. And having a workshop would definitely be better for me!

Reply:
You forget the hard part :D Now you need to convert it back to object and you will need `forEach` or `for`... Anyway it's not purpose to argue about this, it's just whole different approach.